microproxy/protocol/http2.py

```diff
@@ -38,6 +39,8 @@ def decorate(*args):
 
         self.stream = stream
 
+        self.on_pending_events = on_pending_events or on_unhandled_with_type("PENDING_EVENTS")
+        self.on_header_receive = on_header_receive or on_unhandled_with_type("REQUEST_HEADER_RECEIVE")
```

Comment:
The ```on_header_receive``` will not actually sending the header event. Instead, what it will do is just marking the stream as a **pending event stream**.  (I should probably change the ```self.on_header_receive``` to other much proper name. )
So that any other event received after that will be put into the **self.pending_events_of_stream**, and will be send to destination after we have receive the **EndStream** event.